### DISCUSSION TITLE
Live-reload takes too long for reloading when there're a lot of routes.

### DISCUSSION THREAD

**Discussion by ChayaninSuatap** [remix-run/remix#6373] [Q&A]
Is Remix live-reload rebuild source incrementally? From my observation it will take more time to rebuild if there are more files present in app directories (even I edit only one file. It seems the other files were rebuilt too).

For example, initially, when I create a new remix project and edit the initial index file. It takes about 300 ms to rebuild.
![image](https://github.com/remix-run/remix/assets/5347061/fcb35aa3-dd41-4316-9409-9b731c5b62e1)

But when I try copying that index file to create other 100 routes (trying to simulate that the work project is big) the rebuild time now build for 1.2 second.
![image](https://github.com/remix-run/remix/assets/5347061/654db317-340c-4a95-89d1-52be4c409595)

Now this is an extreme example, I copied the index file about 1,000 times. So there are 1,000 routes in the app. It took about 14 seond. (I just add a single character on a page...)
![image](https://github.com/remix-run/remix/assets/5347061/84a26455-9180-4599-9178-a42fe42f9003)

In my working project right now. It took about 4 second for live-reloading. Which is so long consider just changing a class of an element in a file.

I prefer to use Remix. It is great. But this affect development experience so much. Is there any setting can be done. Or can anybody suggest me a workaround on this. Thank you.

**Comment by ChayaninSuatap** [comment]
I am using mantine.dev library in my project. And has to remove [renderToPipeableStream](https://react.dev/reference/react-dom/server/renderToPipeableStream) from **entry.server.tsx**, Not sure if this is the one that affect rebuild time.

**Reply by roopaish** [reply]
I am also using mantine. Btw my entry.server.tsx looks like this. There's no renderToPipelineStream, any idea what to do here? My live-reload takes like 25s sometimes.

```tsx
import { createStylesServer, injectStyles } from '@mantine/remix';
import type { EntryContext } from '@remix-run/node';
import { RemixServer } from '@remix-run/react';
import { renderToString } from 'react-dom/server';

const server = createStylesServer();

export default function handleRequest(
  request: Request,
  responseStatusCode: number,
  responseHeaders: Headers,
  remixContext: EntryContext
) {
  let markup = renderToString(<RemixServer context={remixContext} url={request.url} />);
  responseHeaders.set('Content-Type', 'text/html');

  return new Response(`<!DOCTYPE html>${injectStyles(markup, server)}`, {
    status: responseStatusCode,
    headers: responseHeaders,
  });
}
```

**Reply by ChayaninSuatap** [reply]
My entry.server.tsx looks pretty much like yours. Maybe It's about how mantine is built inside the lib.

**Comment by KwaminaWhyte** [comment]
Same issue here. I have 44 route files and it takes a while for the dev server to reload. As the route files grows it keep getting worse.

**Comment by roopaish** [comment]
I migrated my whole app to next.js(app dir). It's much better.

**Reply by arnaudambro** [reply]
lol, that's very useful for this problem

**Comment by wiljeder** [comment]
I'm doing my first remix project. I only have 4 routes and the live reload is taking between 1 and 2 seconds, which is not a lot, but knowing that as my project grows this will grow by a lot (by the comments above) is demotivating to say the least. And i'm not even using mantine, just a few shadcn components.

I tried changing the node version and creating a remix project from scratch and the reload was fast so i don't really know what's causing it

Edit: with nightly remix dependencies the build time decreased by ~300ms

**Comment by yard2010** [comment]
Is there a way to get sub-1-second reload delay with Remix in 2024? Currently, I have like 4-5 seconds for each reload. If there's no solution maybe change the name to cold reload so people won't have the wrong idea about this feature?
